jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/commands/GetUserTaskCommand.java

```diff
@@ -0,0 +1,72 @@
+package org.jbpm.services.task.commands;
+
+import java.util.List;
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlRootElement;
+
+import org.jbpm.services.task.exception.PermissionDeniedException;
+import org.kie.api.task.model.OrganizationalEntity;
+import org.kie.api.task.model.Task;
+import org.kie.internal.command.Context;
+
+@XmlRootElement(name = "get-user-task-command")
+@XmlAccessorType(XmlAccessType.NONE)
+public class GetUserTaskCommand extends TaskCommand<Task> {
+
+    private static final long serialVersionUID = -3066272693452263188L;
+
+    public GetUserTaskCommand() {
+    }
+
+    public GetUserTaskCommand(String userId,
+                              long taskId) {
+        this.taskId = taskId;
+        this.userId = userId;
+    }
+
+    public Task execute(Context cntxt) {
+        TaskContext context = (TaskContext) cntxt;
+
+        Task task = context.getTaskQueryService().getTaskInstanceById(taskId);
+
+        if (task == null) {
+            throw new IllegalStateException("Unable to find task with id " + taskId);
+        }
+
+        if (!isPotentialOwner(userId,
+                              task,
+                              context)) {
+            throw new PermissionDeniedException("User " + userId + " isn't allowed to see thask " + taskId);
+        }
+
+        return task;
+    }
+
+    protected boolean isPotentialOwner(String userId,
+                                       Task task,
+                                       TaskContext context) {
+        List<String> usersGroup = context.getUserGroupCallback().getGroupsForUser(userId,
+                                                                                  null,
+                                                                                  null);
+        usersGroup.add(userId);
+
+        if (isPotentialOwner(usersGroup,
+                             task.getPeopleAssignments().getPotentialOwners())) {
+            return true;
+        }
+
+        return isPotentialOwner(usersGroup,
+                                task.getPeopleAssignments().getBusinessAdministrators());
+    }
+
+    protected boolean isPotentialOwner(List<String> userGroups,
```

Comment:
name of this method is bit confusing as it is also used for checking business admins